pkg/framework/plugins/defaultevictor/constraints.go

```diff
@@ -0,0 +1,194 @@
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaultevictor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/descheduler/pkg/api"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	frameworktypes "sigs.k8s.io/descheduler/pkg/framework/types"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+func evictionConstraintsForFailedBarePods(evictFailedBarePods bool) constraint {
+	if evictFailedBarePods {
+		klog.V(1).InfoS("Warning: EvictFailedBarePods is set to True. This could cause eviction of pods without ownerReferences.")
+		return func(pod *v1.Pod) error {
+			ownerRefList := podutil.OwnerRef(pod)
+			if len(ownerRefList) == 0 && pod.Status.Phase != v1.PodFailed {
+				return fmt.Errorf("pod does not have any ownerRefs and is not in failed phase")
+			}
+			return nil
+		}
+	}
+	return func(pod *v1.Pod) error {
+		if len(podutil.OwnerRef(pod)) == 0 {
+			return fmt.Errorf("pod does not have any ownerRefs")
+		}
+		return nil
+	}
+}
+
+func evictionConstraintsForSystemCriticalPods(evictSystemCriticalPods bool, priorityThreshold *api.PriorityThreshold, handle frameworktypes.Handle) []constraint {
+	var constraints []constraint
+
+	if !evictSystemCriticalPods {
+		constraints = append(constraints, func(pod *v1.Pod) error {
+			if utils.IsCriticalPriorityPod(pod) {
+				return fmt.Errorf("pod has system critical priority")
+			}
+			return nil
+		})
+
+		if priorityThreshold != nil && (priorityThreshold.Value != nil || len(priorityThreshold.Name) > 0) {
+			thresholdPriority, err := utils.GetPriorityValueFromPriorityThreshold(context.TODO(), handle.ClientSet(), priorityThreshold)
+			if err != nil {
+				klog.Errorf("failed to get priority threshold: %v", err)
+				return constraints
+			}
+			constraints = append(constraints, func(pod *v1.Pod) error {
+				if !IsPodEvictableBasedOnPriority(pod, thresholdPriority) {
+					return fmt.Errorf("pod has higher priority than specified priority class threshold")
+				}
+				return nil
+			})
+		}
+	} else {
+		klog.V(1).InfoS("Warning: EvictSystemCriticalPods is set to True. This could cause eviction of Kubernetes system pods.")
+	}
+
+	return constraints
+}
+
+func evictionConstraintsForLocalStoragePods(evictLocalStoragePods bool) constraint {
+	if !evictLocalStoragePods {
+		return func(pod *v1.Pod) error {
+			if utils.IsPodWithLocalStorage(pod) {
+				return fmt.Errorf("pod has local storage and descheduler is not configured with evictLocalStoragePods")
+			}
+			return nil
+		}
+	}
+	return func(pod *v1.Pod) error { return nil }
```

Comment:
This function will be a no-op and will be invoked every time a pod is checked for eviction. What about to 
change the function signature to return `[]constraint` and `return nil` instead?